baseframe/forms/auto.py

```diff
@@ -54,18 +58,26 @@ def render_redirect(url, code=302):
         return redirect(url, code=code)
 
 
-def render_delete_sqla(obj, db, title, message, success=u'', next=None, cancel_url=None):
+def render_delete_sqla(obj, db, title, message, success=u'', next=None, cancel_url=None, with_response=True):
     if not obj:
         abort(404)
     form = ConfirmDeleteForm()
     if request.method in ('POST', 'DELETE') and form.validate():
+        print ('form', request.form)
         if 'delete' in request.form or request.method == 'DELETE':
             db.session.delete(obj)
             db.session.commit()
             if success:
                 flash(success, 'success')
+            if request.is_xhr:
+                return make_response(jsonify(status='ok', doc=_(u"Deleted"), result={'url': next or url_for('index')}), 303)
```

Comment:
303 response needs a `Location` header. The Ajax handler will replay a GET request to the `Location` URL. Better to respond with 200 here. [This discussion argues for 200](https://stackoverflow.com/questions/6439416/deleting-a-resource-using-http-delete).

Reply:
Fixed